temporalio/worker/_worker.py

```diff
@@ -666,6 +713,31 @@ class WorkerConfig(TypedDict, total=False):
     on_fatal_error: Optional[Callable[[BaseException], Awaitable[None]]]
     use_worker_versioning: bool
     disable_safe_workflow_eviction: bool
+    deployment_config: Optional[WorkerDeploymentConfig]
+
+
+@dataclass
+class WorkerDeploymentConfig:
+    """Options for configuring the Worker Versioning feature.
+
+    WARNING: This is an experimental feature and may change in the future.
+    """
+
+    version: WorkerDeploymentVersion
+    use_worker_versioning: bool
```

Comment:
Hrmm, I wonder if we should default this to `False`. I admit I am unsure what the situations are where you'd set deployment config without wanting to set this to `True`, but it is default false in other SDKs. Regardless, non-blocking...

Reply:
This will default to False, no?

Reply:
Hrmm, do we consider setting `build_id=` as deprecated if they don't want versioning? We'll always have to support the shortcut. I guess it's similar to worker tuning, which arguably there if we always have to support the shortcut we might as well not deprecate its use.

Reply:
Yeah, I think the problem is that from a product perspective they want to de-emphasize build id in favor of the deployment as a whole